Add docs for k8s + otel log insight setup (#254)
pganalyze/pganalyze-docs: log-insights/setup/opentelemetry.mdx

> ---

Author: TODO: add link from https://pganalyze-we-add-selfho-wbo8ie.herokuapp.com/docs/log-insights/setup/self-managed/01_determine_log_location

title: 'Log Insights: Kubernetes with OpenTelemetry'
backlink_href: /docs/log-insights/setup
backlink_title: 'Log Insights: Setup'
---

> If you are running the database in the Kubernetes cluster, you can use the

Reviewer: ```suggestion
If you are running the database in a Kubernetes cluster, you can use the
```

[`db_log_otel_server`](/docs/collector/settings#self-managed-servers) setting to
have the collector listen for OpenTelemetry log data model messages.

> When this is specified, the collector will run the HTTP server behind the scenes

Reviewer: ```suggestion
When this is specified, the collector will run an HTTP server behind the scenes
```

on the specified port to receive logs at the `/v1/logs` endpoint.
This should be a local address like `0.0.0.0:4318` and it's recommended to use

> an unprivileged port number (1024 and up) to avoid running the HTTP server as

Reviewer: (Here 'the' is correct, because you're referencing the specific server that the collector is running.)

root.

> **Note:** This is still beta and currently only supporting CloudNativePG logs,

Author: I guess we don't really say "beta", but I wanted to add some hint that this is still pretty much wip

> **Note:** This is currently only supporting CloudNativePG logs, with the jsonlog
> format (available with Postgres 15 or above).

Reviewer: ```suggestion
**Note:** This currently only supports CloudNativePG logs, with the jsonlog
format (available with Postgres 15 or above).
```

Separately, should we explicitly reference the [log_destination](https://www.postgresql.org/docs/current/runtime-config-logging.html#GUC-LOG-DESTINATION) setting here? And maybe link it? Or is that not helpful in this context?

Author: Actually, after the research, turned out that this is not correct. Well, "This currently only supports CloudNativePG logs" is correct, but CloudNativePG actually uses `log_destination = 'csvlog'` (https://cloudnative-pg.io/documentation/1.23/postgresql_conf/) then they convert to JSON style log: https://cloudnative-pg.io/documentation/1.23/logging/#postgresql-log

I'll update to drop the latter part. Thanks for leaving comment on this one, this actually has been confusing me for a while so I'm glad that now I know what exactly CloudNativePG does and what/how we support.
FYI @lfittl as we chatted about this yesterday a bit

### Using with a pganalyze collector Helm chart

If you are deploying the pganalyze collector using a Helm chart, you can update
the values.yaml file to create a service for this HTTP server.
In values.yaml file, update the section of `service`:

```yaml
service:
  create: true
  name: pganalyze-collector-otel-service
  type: ClusterIP
  port: 4318
  targetPort: 4318
```

This will create a service with the name `pganalyze-collector-otel-service`.
Assuming you use [Fluent Bit](https://fluentbit.io/) as a telemetry agent,
you can specify the created service as an OpenTelemetry HTTP endpoint, using the
[OpenTelemetry output plugin](https://docs.fluentbit.io/manual/pipeline/outputs/opentelemetry):

```
  [OUTPUT]
    name  opentelemetry
    match kube.*postgres*
    host  pganalyze-collector-otel-service
    port  4318
```

> With this flow, Fluent Bit tails logs of Postgres pods, sends out logs to the
> pganalyze collector's OpenTelemetry HTTP endpoint (via `pganalyze-collector-otel-service`),
> then the collector will process logs.

Reviewer: ```suggestion
With this flow, Fluent Bit tails the logs of Postgres pods, sends the logs to the
pganalyze collector's OpenTelemetry HTTP endpoint (via `pganalyze-collector-otel-service`),
and the collector processes the logs.
```
Some wording adjustments here: mostly I think we should keep the same verb tense.
